pkg/apply/solver/solver_test.go

```diff
@@ -429,6 +429,12 @@ func TestTaskQueueBuilder_AppendApplyWaitTasks(t *testing.T) {
 							expTsk.Ids, actWaitTask.Ids)
 					}
 					assert.Equal(t, taskrunner.AllCurrent, actWaitTask.Condition)
+					if tc.options.ReconcileTimeout != 0 {
+						assert.Equal(t, tc.options.ReconcileTimeout, expTsk.Timeout)
+						assert.Equal(t, expTsk.Timeout, actWaitTask.Timeout)
+					} else {
+						assert.Equal(t, tc.options.ReconcileTimeout, actWaitTask.Timeout)
```

Comment:
This else seems redundant. Isn't this always being tested?

Reply:
No. If `tc.Options.ReconcileTimeout` is not set, `WaitTask.Timeout` will be zero. We can either test `assert.Equal(t, tc.options.ReconcileTimeout, actWaitTask.Timeout)` or change the `WaitTask.Timeout` in `tc.expectedTasks` from `1*time.Second` to 0.

Reply:
What i mean is that because of the transitive property `tc.options.ReconcileTimeout == expTsk.Timeout && expTsk.Timeout == actWaitTask.Timeout` means that `tc.options.ReconcileTimeout == actWaitTask.Timeout`. So the 2nd line in the if statement is equivalent to the else statement. We could just move it out, right?

```
if tc.options.ReconcileTimeout != 0 {
    assert.Equal(t, tc.options.ReconcileTimeout, expTsk.Timeout)
}
assert.Equal(t, tc.options.ReconcileTimeout, actWaitTask.Timeout)
```